Code/Tools/AssetProcessor/native/unittests/AssetProcessorManagerUnitTests.cpp

```diff
@@ -2963,4 +2963,134 @@ namespace AssetProcessor
 
         assetBuilderInfoHandler.BusDisconnect();
     }
+
+    // Helper function, processes assets and blocks until complete
+    void AssetProcessorManagerUnitTests::ProcessAssetBlockUntilComplete(QString& assetToProcess)
+    {
+        m_processResults.clear();
+        QMetaObject::invokeMethod(
+            m_assetProcessorManager.get(), "AssessModifiedFile", Qt::QueuedConnection, Q_ARG(QString, assetToProcess));
+        EXPECT_TRUE(BlockUntil(m_idling, 5000));
+        QCoreApplication::processEvents(QEventLoop::AllEvents);
+
+        AssetProcessorManagerUnitTestUtils::SortAssetToProcessResultList(m_processResults);
+    }
+    #if 1
+    // This test verifies the fingerprint clearing command causes assets to reprocess.
+    // It does this by processing assets once to setup. Then processes them a second time, to verify
+    // they don't reprocess due to identical fingerprints. After that, it clears the fingerprint from the DB,
+    // and processes a final time, to verify the assets process because of the fingerprint change.
+    TEST_F(AssetProcessorManagerUnitTests, FingerprintClearRequest_ClearsFingerprints_AssetsReprocess)
```

Comment:
will this unit test be safe to run on an Android and/or iOS?

Reply:
We don't run the asset processor on Android/iOS yet. I don't think we have plans to, and if someone wanted to do so, it would take a lot of work to make the tests work on that platform.